config/ProviderConfig.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+class ProviderConfig {
+  constructor() {
+    this.google = {
```

Comment:
I really wanted these to be versioned, as these providers seem to upgrade their OAuth endpoints regularly with versioning. It would be easy to have the google:{ v3:{} } object for instance, and then just add v4 support later down the line while keeping backwards compatibility. What do you think?